### PR TITLE
added pagination

### PR DESCRIPTION
**Pull request by amir-ben-shimol**

### CODE REVIEW

**Comment by roy-shiloh** [comment]
Please add instructions for starting dev mode if you want me to reviewSent from my iPhoneOn 18 May 2023, at 14:46, beargreenholtz ***@***.***> wrote:﻿
@beargreenholtz requested review from @Hayota-tech/developers on: #56 added pagination.

File: src/components/ui/SProjectCard/SProjectCard.view.tsx [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
 	readonly projectName: string;
 	readonly year: string;
 	readonly media: string | undefined;
 	readonly category: string | undefined;
+	readonly keyNum: number;
+	readonly projectNumber?: number;
 }
 
 const SProjectCardView: React.FC<IProps> = (props: React.PropsWithChildren<IProps>) => {
+	let container;
```
Comment (amir-ben-shimol) [inline src/components/ui/SProjectCard/SProjectCard.view.tsx:17]: check the state inside of `concatClasses`

Look for prevues use that i did

File: src/components/containers/Works/Works.tsx [hunk 2]
```diff
@@ -12,6 +12,17 @@ const Works: React.FC<IProps> = () => {
...
 	const [categoriesListState, setCategoriesListState] = useState<ICategory[]>([]);
 	const [typesListState, setTypesListState] = useState<IType[]>([]);
+	const [projectNumberState, setProjectNumberState] = useState<number>(10);
+
+	const handleScroll = (e: React.UIEvent<HTMLElement>) => {
+		const target = e.target as HTMLElement;
+		const bottom = target.scrollHeight - target.scrollTop === target.clientHeight;
+
+		if (bottom) {
+			console.log('reached');
```
Comment (amir-ben-shimol) [inline src/components/containers/Works/Works.tsx:22]: remove

File: src/components/containers/Works/Works.tsx [hunk 3]
```diff
@@ -28,17 +39,21 @@ const Works: React.FC<IProps> = () => {
...
+				.then((response: AxiosResponse) => {
+					setProjectsListState(() => response.data.data);
+				});
+		}
 	}, [backendApi]);
 
 	return (
 		<WorksView
+			projectNumber={projectNumberState}
+			handleScroll={handleScroll}
```
Comment (amir-ben-shimol) [inline src/components/containers/Works/Works.tsx:56]: functions should appear last

File: src/components/containers/Works/Works.view.tsx [hunk 4]
```diff
@@ -11,16 +11,19 @@ interface IProps {
 	readonly projectsList: IProjectsList[];
 	readonly categoriesList: ICategory[];
 	readonly typesList: IType[];
+	readonly handleScroll: any;
```
Comment (amir-ben-shimol) [inline src/components/containers/Works/Works.view.tsx:14]: never use any type

File: src/components/containers/Works/Works.view.tsx [hunk 5]
```diff
@@ -11,16 +11,19 @@ interface IProps {
...
+	readonly projectNumber?: number;
```
Comment (amir-ben-shimol) [inline src/components/containers/Works/Works.view.tsx:15]: optional `?` is redundant here

File: src/components/containers/Main/Projects/Projects.view.tsx [hunk 6]
```diff
@@ -22,6 +22,7 @@ interface IProps {
 	readonly onSelectType: (_: IType) => void;
 	readonly onSelectAllTypes: () => void;
 	readonly onNavigateToProject: (_: string) => void;
+	readonly projectNumber?: number;
```
Comment (amir-ben-shimol) [inline src/components/containers/Main/Projects/Projects.view.tsx:25]: functions should be last

File: src/components/ui/SProjectCard/SProjectCard.tsx [hunk 7]
```diff
@@ -7,11 +7,15 @@ interface IProps {
 	readonly year: string;
 	readonly media: string | undefined;
 	readonly category: string | undefined;
+	readonly keyNum: number;
```
Comment (amir-ben-shimol) [inline src/components/ui/SProjectCard/SProjectCard.tsx:10]: change to `index`

File: src/components/ui/SProjectCard/SProjectCard.tsx [hunk 8]
```diff
@@ -7,11 +7,15 @@ interface IProps {
...
+	readonly projectNumber?: number;
```
Comment (amir-ben-shimol) [inline src/components/ui/SProjectCard/SProjectCard.tsx:11]: change to `maxProjectsVisibleIndex`

File: src/components/ui/SProjectCard/SProjectCard.tsx [hunk 9]
```diff
@@ -7,11 +7,15 @@ interface IProps {
...
 }
 
 const SProjectCard: React.FC<IProps> = (props: React.PropsWithChildren<IProps>) => {
 	return (
 		<SProjectCardView
+			keyNum={props.keyNum}
+			projectNumber={props.projectNumber}
```
Comment (amir-ben-shimol) [inline src/components/ui/SProjectCard/SProjectCard.tsx:18]: change to `maxProjectsVisibleIndex`

File: src/components/ui/SProjectCard/SProjectCard.tsx [hunk 10]
```diff
@@ -7,11 +7,15 @@ interface IProps {
...
 	readonly media: string | undefined;
 	readonly category: string | undefined;
+	readonly keyNum: number;
+	readonly projectNumber?: number;
 }
 
 const SProjectCard: React.FC<IProps> = (props: React.PropsWithChildren<IProps>) => {
 	return (
 		<SProjectCardView
+			keyNum={props.keyNum}
```
Comment (amir-ben-shimol) [inline src/components/ui/SProjectCard/SProjectCard.tsx:17]: change to `index`

**Review by amir-ben-shimol — COMMENTED** [review]
Functionality doesnt work

Please check again and fix the comments

@beargreenholtz

**Pull request merged**